src/utils/transforms.cljs

```diff
@@ -21,3 +24,37 @@
   [json]
   (when-not (= json "undefined")
     (try (.parse js/JSON json) (catch js/Error _ (when (string? json) json)))))
+
+;; Functions took from reagent and modified to also transform vectors contained in styles
+
+(def ^:private js-val? reagent.impl.util/js-val?)
+(def ^:private named? reagent.impl.util/named?)
+(def ^:private cached-prop-name reagent.impl.template/cached-prop-name)
+(declare styles-with-vectors)
+
+(defn ^:private kv-conv
+  [o k v]
+  (doto o
+    (oops/gobj-set (cached-prop-name k) (styles-with-vectors v))))
+
+(defn styles-with-vectors
+  [x]
+  (cond (js-val? x) x
+        (named? x)  (name x)
+        (map? x)    (reduce-kv kv-conv #js {} x)
+        (vector? x) (to-array (map styles-with-vectors x))
+        (coll? x)   (clj->js x)
+        (ifn? x)    (fn [& args]
```

Comment:
I'm not sure, I think no.

I just kept them because I didn't want to differ too much from the Reagent implementation.

Reply:
In this case I think it's safe to ignore the `ifn?` check because we know `styles-with-vectors` is called only for style maps, and as far as you and me remember, there are no cases of style maps with values as functions. If we don't do this now, probably nobody will bother to do in future PRs. It's just nice in terms of performance to have less conditions.

Reply:
Done!